src/ctx.rs

```diff
@@ -2,30 +2,100 @@ use crate::include::stdint::uint16_t;
 use crate::include::stdint::uint32_t;
 use crate::include::stdint::uint64_t;
 use crate::include::stdint::uint8_t;
+use std::mem;
+
+pub trait Alias {
+    fn set(&mut self, val: u64);
+}
 
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub union alias8 {
     pub u8_0: uint8_t,
 }
 
+impl Alias for alias8 {
+    fn set(&mut self, val: u64) {
+        self.u8_0 = val as u8;
+    }
+}
+
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub union alias16 {
     pub u16_0: uint16_t,
     pub u8_0: [uint8_t; 2],
 }
 
+impl Alias for alias16 {
+    fn set(&mut self, val: u64) {
+        self.u16_0 = val as u16;
+    }
+}
+
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub union alias32 {
     pub u32_0: uint32_t,
     pub u8_0: [uint8_t; 4],
 }
 
+impl Alias for alias32 {
+    fn set(&mut self, val: u64) {
+        self.u32_0 = val as u32;
+    }
+}
+
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub union alias64 {
     pub u64_0: uint64_t,
     pub u8_0: [uint8_t; 8],
 }
+
+impl Alias for alias64 {
+    fn set(&mut self, val: u64) {
+        self.u64_0 = val;
+    }
+}
+
+#[inline]
+pub unsafe fn set_ctx_rep1<T: Alias>(buf: *mut u8, off: isize, val: u64) {
+    let buf = buf.offset(off);
+    let buf = buf.cast::<T>();
+    // debug_assert!(buf.is_aligned()); // #![feature(pointer_is_aligned)]
+    debug_assert!(buf.cast::<u8>() as usize & mem::align_of::<T>() - 1 == 0);
```

Comment:
This alignment debug check doesn't work when using `T` that are not extra aligned.  That's why I used `Align16` and `Align32`, as the C code expects it to be aligned to that in those cases.

Reply:
Ah, okay. I'll remove it here then since it's not doing anything.